### PR TITLE
feat: 🎸 Proposal: Decoding JSON in RealityComposerStrategy

### PR DESCRIPTION
**Pull request by pgobriensap**
BREAKING CHANGE: 🧨 Simplified RealityComposerStrategy Init/Renamed Errors

- **Proposal**: Init to accept json Data
- To do this I needed a Concrete Implementation of CardItemModel to conform to Decodable.
- As a result when using JSON to load data the developer is constrained to `DefaultCardItem`
- Open to better naming convention

- If proposal is Accepted I can create an Encodable Init when my bandwidth increases.

As for breaking changes, I experienced some of these errors and they weren't clear without Error appended. The init for object anchors was just for convenience and didn't add anything. Documentation can clarify this.

### CODE REVIEW

**Comment by pgobriensap** [comment]
After thinking about it I could have the CardItem generic in the Strategy Conform to Decodable. This would cause the strategy to need an explicit declaration of the CardItem Type. However perhaps a DefaultCardItem is still appealing in case the developers only want the default content. So they don't have to worry about creating a custom Decoding for their concrete.

**Review by MarcoEidinger — CHANGES_REQUESTED** [review]
You did not commit `Tests.json` and therefore the example app cannot be build out-of-the-box

File: Sources/FioriARKit/ARCards/Models/CardItemModel.swift [hunk 1]
```diff
@@ -28,3 +28,53 @@ public protocol ActionTextComponent {
 public protocol IconComponent {
     var icon_: Image? { get }
 }
+
+internal struct CodableCardItem: Codable {
```
Comment (MarcoEidinger) [inline Sources/FioriARKit/ARCards/Models/CardItemModel.swift:32]: not used so can be deleted

File: Sources/FioriARKit/ARCards/Models/CardItemModel.swift [hunk 2]
```diff
@@ -28,3 +28,53 @@ public protocol ActionTextComponent {
...
+    var id: String
+    var title: String
+    var descriptionText: String?
+    var detailImage: Data?
+    var actionText: String?
+    var icon: String?
+}
+
+public struct DefaultCardItem: CardItemModel {
```
Comment (MarcoEidinger) [inline Sources/FioriARKit/ARCards/Models/CardItemModel.swift:41]: Proposing `DecodableCardItem` as name

**Comment by pgobriensap** [comment]
Yeah, good catch. It was just a reference to the file on my disk and not added into the project directory. It has been pushed.

I also decided to use base64 Encoding for the image. Should there also be an alternative where the image could be a string to a file path of an image?

**Review by MarcoEidinger — COMMENTED** [review]
> I also decided to use base64 Encoding for the image. Should there also be an alternative where the image could be a string to a file path of an image?

Let's leave it for base64 only

> So they don't have to worry about creating a custom Decoding for their concrete.

Yeah, I think that's the main benefit here. In the end they can decode their custom type and pass it as long as it conforms to `CardItemModel`

File: Sources/FioriARKit/ARCards/Models/CardItemModel.swift (hunk 2 above)
Comment (pgobriensap) [reply Sources/FioriARKit/ARCards/Models/CardItemModel.swift:41]: I'm considering the usefulness of also having it encodable but focused on decoding the json first. For now I can't think of a reason the content will change for us to provide a new json as a convenience for the developer. I'm also unsure of how to encode the `Image` back to base64.

If this is not necessary then I think 'DecodableCardItem' is reasonable.

Comment (MarcoEidinger) [reply Sources/FioriARKit/ARCards/Models/CardItemModel.swift:41]: Please use either `DecodableCardItem` or `CodableCardItem` as name but not `DefaultCardItem`

**Review by MarcoEidinger — CHANGES_REQUESTED** [review]
please delete unused code and do renaming

File: Sources/FioriARKit/ARCards/Models/CardItemModel.swift (hunk 2 above)
Comment (pgobriensap) [reply Sources/FioriARKit/ARCards/Models/CardItemModel.swift]: Renamed to DecodableCardItem

**Review by MarcoEidinger — APPROVED** [review]

**Pull request merged**